tests/hypervisor/test_esx.py

```diff
@@ -1432,6 +1433,54 @@ def test_post_large_json_to_rhsm(self, register_data, ssh_host):
         else:
             logger.warning("Failed to post json to satellite")
             logger.warning(output)
+    
+    @pytest.mark.tier2
+    @pytest.mark.notStage
+    def test_run_in_FIPS_mode(self, function_hypervisor, virtwho, ssh_host):
+        """
+        :title: virt-who: esx: test run virt-who in FIPS mode
+        :id: 2a05a366-6b92-456a-89ca-c7576491c0bb
+        :caseimportance: High
+        :tags: hypervisor,esx,tier2
+        :customerscenario: false
+        :upstream: no
+        :steps:
+            1. Enable FIPS mode on the host
+            2. Reboot the host
+            3. Run virt-who service
+            4. Check the virt-who service status
+            5. Disable FIPS mode on the host
+        :expectedresults:
+            1. Succeeded to enable FIPS mode on the host
+        """
+        timeout = 300
+        interval = 5
+        
+        ssh_host.runcmd("fips-mode-setup --enable")
+        ssh_host.runcmd("sudo reboot")
+        start_time = time.time()
+        # 等待几秒，以便服务器开始重启过程
+        time.sleep(interval)
+        while not is_host_responsive(ssh_host.host):
+            elapsed_time = time.time() - start_time
+            if elapsed_time > timeout:
+                assert False, f"Timeout reached after {timeout} seconds!"
+            print(f"Waiting for server to come back online... Elapsed time: {int(elapsed_time)} seconds.")
+            time.sleep(interval)
+        
+        _, stdout = ssh_host.runcmd("cat /proc/sys/crypto/fips_enabled")
+        assert("1" in stdout)
+        
+        result = virtwho.run_service()
+        print(result)
+        assert (
+            result["error"] == 0
+            and result["send"] == 1
+            and result["thread"] == 1
+        )
+        
+        ssh_host.runcmd("fips-mode-setup --disable")
```

Comment:
to confirm we change back to disable mode after this case, maybe we need to use the `try...finally..`. like case: https://github.com/VirtwhoQE/virtwho-test/blob/main/tests/package/test_upgrade_downgrade.py#L33
what do you think of this?

Reply:
Maybe I think no need to use try...finally in this code. I don't know what to do in finally... sentences.

Reply:
we need to change back to FIPS disable mode, so we should add the `fips-mode-setup --disable` to finally